File: html/old/backup/sacor2_2008/tiles.py
#!/usr/bin/env python
# SACO Second Round 2008
# Problem: Tiles
# Author:  Marco Gallotta

N = input("Enter N: ")
count = [0,0,1,1]
for i in xrange(4, N+1):
    count.append((count[i-2] + count[i-3]) % 1007)
print count[N]

File: html/old/backup/sacor2_2008/tiles.java
// SACO Second Round 2008
// Problem: Tiles
// Author: Timothy Stranex

import java.io.*;
public class tiles
{
    public static void main(String[] args)
        throws IOException
    {
       BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
       System.out.print("Enter N: ");
       int N = Integer.parseInt(in.readLine());
       long[] W = new long[N+1];
       W[0] = 1;
       W[1] = 0;
       W[2] = 1;
       for (int i=3; i<=N; i++) {
           W[i] = (W[i-2] + W[i-3]) % 1007;
       }
       System.out.println(W[N]);
    }
}
